File: ECarManiac/ECarManiac/Models/Car.cs
using System.Security;
using Microsoft.AspNetCore.Mvc.ModelBinding;
using Microsoft.VisualBasic;

namespace ECarManiac.Models
{
    public class Car
    {
        public int Id { get; set; }
        public string Brand { get; set; }
        public string Model { get; set; }
        public string SubType { get; set; }
        public int Year { get; set; }
        public string Country { get; set; }
        public string BodyType { get; set; }
        public string BatteryType { get; set; }
        public int BatteryCapacity { get; set; }
        public int RealRange { get; set; }
        public int ChargingSpeed { get; set; }
        public int Performance { get; set; }
        public int Torque { get; set; }
        public decimal Acceleration { get; set; }
        public int MaxSpeed { get; set; }
        public int Trunk { get; set; }
        public string Frunk { get; set; }

        public Car(int id, string brand, string model, string sub_type, int year, string country, string body_type, string battery_type, int battery_capacity,
            int real_range, int charging_speed, int performance, int torque, decimal acceleration, int max_speed, int trunk, string frunk)
        {
            Id = id;
            Brand = brand;
            Model = model;
            SubType = sub_type;
            Year = year;
            Country = country;
            BodyType = body_type;
            BatteryType = battery_type;
            BatteryCapacity = battery_capacity;
            RealRange = real_range;
            ChargingSpeed = charging_speed;
            Performance = performance;
            Torque = torque;
            Acceleration = acceleration;
            MaxSpeed = max_speed;
            Trunk = trunk;
            Frunk = frunk;
        }

        public override string ToString()
        {
            return new string($"{Brand} {Model} {SubType}");
        }
    }
}

File: e-car-maniac-fe/src/Pages/Car.js
import { useEffect, useState, useCallback } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import Fetch from '../Other/Fetch';

const Car = () => {
    const navigate = useNavigate();
    const params = useParams();
    const [loading, setLoading] = useState(true);
    const [car, setCar] = useState(false);

    useEffect(() => {
        const dataFetch = async () => {
            const data = await Fetch("get", params.id, "");
            setCar(data);
            setLoading(false);
        }
        dataFetch();
    }, []);

    const deleteCar = useCallback(() => {
        Fetch("delete", params.id, "");
        navigate("/");
    }, []);

    return (
        <>
            <h1>Car Page</h1>
            {loading ?
                <div class="spinner-border" role="status">
                    <span class="visually-hidden">Loading...</span>
                </div>
                : car ? (
                    <div>
                        <h2>{car.brand} {car.model} {car.subType === "null" ? "" : car.subType}</h2>
                        <ul class="list-group">
                            <li class="list-group-item">Vintage: {car.year}</li>
                            <li class="list-group-item">Manufactured in: {car.country}</li>
                            <li class="list-group-item">Body Type: {car.bodyType}</li>
                            <li class="list-group-item">Battery Type: {car.batteryType}</li>
                            <li class="list-group-item">Battery Capacity: {car.batteryCapacity} kWh</li>
                            <li class="list-group-item">Real Range: {car.realRange} km</li>
                            <li class="list-group-item">Max. Charging Speed: {car.chargingSpeed} kW</li>
                            <li class="list-group-item">Power: {car.performance} hp</li>
                            <li class="list-group-item">Torque: {car.torque} Nm</li>
                            <li class="list-group-item">0 - 100 km/h: {car.acceleration} s</li>
                            <li class="list-group-item">Max. Speed: {car.maxSpeed} km/h</li>
                            <li class="list-group-item">Trunk space: {car.trunk} l</li>
                            <li class="list-group-item">Frunk space: {car.frunk === "null" ? 0 : car.Frunk} l</li>
                        </ul>
                        <button type="button" class="btn btn-primary" onClick={() => navigate(-1)}>Go Back</button>
                        <button type="button" class="btn btn-danger" onClick={deleteCar}>Delete</button>
                    </div>
                ) : (
                    <div>
                        <h2>
                            There isn't a car with this id
                        </h2>
                    </div>
                )}
        </>    
    );
}

export default Car;